driver-core/src/main/com/mongodb/internal/operation/ChangeStreamOperation.java

```diff
@@ -325,7 +325,8 @@ public boolean getRetryReads() {
                         (AggregateResponseBatchCursor<RawBsonDocument>) wrapped.execute(binding);
                 return new ChangeStreamBatchCursor<T>(ChangeStreamOperation.this, cursor, binding,
                         setChangeStreamOptions(cursor.getPostBatchResumeToken(), cursor.getOperationTime(),
-                                source.getServerDescription().getMaxWireVersion(), cursor.isFirstBatchEmpty()));
+                                source.getServerDescription().getMaxWireVersion(), cursor.isFirstBatchEmpty()),
+                        source.getServerDescription().getMaxWireVersion());
```

Comment:
I think we discussed this in the chat: it's not sufficient to check the connection source's max wire version, since it could be stale.  We need the max wire version from an actual connection.  So we need to get a Connection from the ConnectionSource here, and then use a SingleConnectionBinding when calling `wrapped.execute(binding)`.

I just had a peek and there is actually both a SingleConnectionBinding and a SingleConnectionReadBinding.  Neither of them are actually used yet in production code (only test code).  I'm not sure why we have both and if we can consolidate them, or whether either are sufficiently tested.  So you'll have to check that out when you make this change.

Reply:
As discussed, we decided to use the max wire version from `AggregateResponseBatchCursor`.